File: 2019/day_01.py
from utils import aoc_comm
import os
import re

# --- update day/ year for each challenge
settings = {
    'day': 1,
    'year': 2019,
    'cookie-path': os.path.realpath('../aoc_cookie.json')
}

# -- OTHER LIBs that might help while coding the soultions
from collections import Counter, defaultdict
from math import *
import functools
import itertools


def parse_input(inp_content):
    inp_content = inp_content.strip()
    # add further input processing here..
    inp_content = [int(e) for e in inp_content.split("\n")]
    return inp_content


def fuel(n):
    return max(floor(n / 3) - 2, 0)

    
@aoc_comm(settings, level=1)
def solve_l1(input_str):  # input data will be passed to this as string
    inp = parse_input(input_str)

    ans = 0
    for ee in inp:
        ans += fuel(ee)

    return ans  # if 'ans' is None answer won't be submitted, else it will submit after confirmation


@aoc_comm(settings, level=2)
def solve_l2(input_str):
    inp = parse_input(input_str)

    ans = 0
    for ee in inp:
        f = fuel(ee)
        while f != 0:
            ans += f
            f = fuel(f)
    return ans


def main():
    # create_test_file(settings, int(input("Test file for level: ")), input("Test result: ").strip())
    solve_l1()
    solve_l2()


if __name__ == '__main__':
    main()

File: 2025/day_01.ts
import { solve } from '../utils';

const YEAR = 2025;
const DAY = 1;

function parseInput(input: string): any {
    return input.split('\n').map(l =>
        [l[0], Number(l.slice(1))]
    );
}


function part1(input: string): any | null {
    let inp = parseInput(input);
    let s = 50;
    let c = 0;
    for (let [dir, val] of inp) {
        s = (s + (dir === "L" ? -val : val)) % 100;
        c += s === 0 ? 1 : 0;
    }

    return c;
}

function part2(input: string): any | null {
    let inp = parseInput(input);
    let s = 50;
    let c = 0;
    for (let [dir, val] of inp) {
        while (val-- > 0) {
            s = ((s + (dir === "L" ? -1 : 1)) % 100 + 100) % 100;
            c += s === 0 ? 1 : 0;
        }
    }

    return c;
}

solve(part1, YEAR, DAY, 1);
solve(part2, YEAR, DAY, 2);